packages/dataviews/src/dataviews.tsx

```diff
@@ -92,13 +100,17 @@ export default function DataViews< Item extends AnyItem >( {
 	}
 	const [ openedFilter, setOpenedFilter ] = useState< string | null >( null );
 
-	const onSetSelection = useCallback(
-		( items: Item[] ) => {
-			setSelection( items.map( ( item ) => getItemId( item ) ) );
-			onSelectionChange( items );
-		},
-		[ setSelection, getItemId, onSelectionChange ]
-	);
+	const prevSelection = usePrevious( selection );
+
+	useEffect( () => {
```

Comment:
Instead of an effect, what do you think of keeping exactly the other changes but instead of having an effect we create a callback with exactly the same signature as setSelection so all the other changes are valid, but that also calls the onSelectionChange inside and not only the setSelection? It would remove the need for the effect.

Reply:
It's a bit less clean. Previously it was using a callback. If we move this all in a callback, it will pretty much re-render all the time based on these dependencies. I guess I can check if that callback was really needed in the first place.

Why do you not like the effect? Cause it could be triggered from outside selection changes? I think that would be expected?

Reply:
> Why do you not like the effect? Cause it could be triggered from outside selection changes? I think that would be expected?

Yah, the main difference with using an effect is that even if the consumer is the one doing the selection change the consumer will still get a callback on the onSelectionChange that may be unexpected. I'm not sure what is the good practice on these cases.

Reply:
@jorgefilipecosta I changed it